entity_parser.go

```diff
@@ -365,20 +368,48 @@ func parseETLTag(tag string) (string, ETLState, error) {
 	return fullETLTag, etlState, nil
 }
 
+// parseTTLTag functions parses DOSA "ttl" tag
+func parseTTLTag(tag string) (string, time.Duration, error) {
+	fullTTLTag := ""
+	ttlTag := ""
+	matches := ttlPattern0.FindStringSubmatch(tag)
+	if len(matches) == 2 {
+		fullTTLTag = matches[0]
+		ttlTag = matches[1]
+	}
+
+	if len(matches) == 0 {
```

Comment:
nit: I'd put this if block before the `len(matches) == 2` since it's the common case.